Assets/Scripts/SelectionManager.cs

```diff
@@ -977,6 +977,30 @@ public void SetSnappingGridSize(int snapIndex)
             }
         }
 
+        public Quaternion QuantizeAngle(Quaternion rotation)
+        {
+            if (SnappingAngle == 0) return Quaternion.identity;
+            float round(float val) { return Mathf.Round(val / SnappingAngle) * SnappingAngle; }
+            Vector3 euler = rotation.eulerAngles;
+            float y = euler.y;
+            euler = new Vector3(round(euler.x), 0, round(euler.z));
+            rotation = Quaternion.Euler(euler);
+            rotation *= Quaternion.Euler(0, round(y), 0);
+            return rotation;
```

Comment:
Would this benefit from a comment explaining why it's now done in two stages (rotate x/z, then rotate y), rather than the previous `euler = new Vector3(round(euler.x), round(euler.y), round(euler.z));`, or is this obvious to people who know what quaternions are?

Reply:
Good point. I'm explained as best I can.